horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/RunServiceImpl.java

```diff
@@ -752,7 +752,8 @@ public RunsSummary listAllRuns(String query, boolean matchAll, String roles, boo
          RunsSummary summary = new RunsSummary();
          // TODO: total does not consider the query but evaluating all the expressions would be expensive
          summary.total = trashed ? RunDAO.count() : RunDAO.count("trashed = false");
-         summary.runs = runs.stream().map(this::createSummary).collect(Collectors.toList());
+         summary.runs = new ArrayList<>(runs.size());
+         runs.forEach(r -> summary.runs.add(createSummary(r)));
```

Comment:
please can we leave the original implementation. Its execution was parallelizable and the code is more readable

Reply:
@whitingjr please can you take a look at this